### PR TITLE
Typed Builder implementation Part 2: Sequences

### PR DESCRIPTION
**Pull request by Kukovec**
- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] Documentation added for any new functionality
- [ ] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

**Review by konnov — APPROVED** [review]
Looks nice and simple!

File: tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/signatures/SeqOperSignatures.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    // (Seq(t), Int, Int) => Seq(t)
+    val subseqSig = subseq -> { _: Int =>
+      val t = varPool.fresh
+      OperT1(Seq(SeqT1(t), IntT1(), IntT1()), SeqT1(t))
+    }
+
+    // (Seq(t), (Seq(t)) => Bool) => Seq(t)
+    // selectseq is rewired away and should not be created
```
Comment (thpani) [inline tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/signatures/SeqOperSignatures.scala:57]: What does "rewired away" mean?

**Review by thpani — APPROVED** [review]
👍

File: tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/signatures/SeqOperSignatures.scala (hunk 1 above)
Comment (Kukovec) [reply tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/signatures/SeqOperSignatures.scala:57]: https://github.com/informalsystems/apalache/blob/d7c22d1d811d8bce268ce43606283ef0baafef26/tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/TlaSeqOper.scala#L58-L60

Comment (thpani) [reply tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/signatures/SeqOperSignatures.scala:57]: Then this is clearer:
```suggestion
    // selectseq is rewired in __rewire_sequences_in_apalache.tla and should not be created
```

Comment (Kukovec) [reply tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/signatures/SeqOperSignatures.scala:57]: Ok, I'll add that in the next pr

**Pull request merged**